### PR TITLE
Kog Operator

### PR DESCRIPTION
**Pull request by SergeRadinovich**

### CODE REVIEW

File: deploy/crds/k8s_v1alpha1_iofog_crd.yaml [hunk 1]
```diff
@@ -10,10 +10,105 @@ spec:
...
+            object represents. Servers may infer this from the endpoint the client
+            submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#types-kinds'
+          type: string
+        metadata:
+          type: object
+        spec:
+          properties:
+            microservices:
+              items:
+                properties:
```
Comment (baghbidi) [inline deploy/crds/k8s_v1alpha1_iofog_crd.yaml]: Don't we need volume mappings?

File: deploy/crds/k8s_v1alpha2_kog_crd.yaml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+          type: object
+        spec:
+          properties:
+            connectors:
+              properties:
+                image:
+                  type: string
+              type: object
+            controlPlane:
+              description: 'INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
```
Comment (baghbidi) [inline deploy/crds/k8s_v1alpha2_kog_crd.yaml]: Is this needed?

File: deploy/operator.yaml [hunk 3]
```diff
@@ -76,12 +30,10 @@ spec:
       labels:
         name: iofog-operator
     spec:
-      nodeSelector:
-        node-role.kubernetes.io/master: ""
       serviceAccountName: iofog-operator
       containers:
         - name: iofog-operator
-          image: iofog/iofog-operator:0.1.0
+          image: gcr.io/focal-freedom-236620/operator:serge
```
Comment (baghbidi) [inline deploy/operator.yaml]: ?

File: deploy/crds/k8s_v1alpha1_iofog_crd.yaml (hunk 1 above)
Comment (SergeRadinovich) [reply deploy/crds/k8s_v1alpha1_iofog_crd.yaml]: This is auto generated using operator sdk. I haven't changed the code that would change this auto generated yaml

File: deploy/crds/k8s_v1alpha2_kog_crd.yaml (hunk 2 above)
Comment (SergeRadinovich) [reply deploy/crds/k8s_v1alpha2_kog_crd.yaml]: Just auto gen again

File: deploy/crds/k8s_v1alpha1_iofog_crd.yaml (hunk 1 above)
Comment (SergeRadinovich) [reply deploy/crds/k8s_v1alpha1_iofog_crd.yaml]: this is commented out - is this intentional?
#      volume-mappings:
#        - hostDestination: "/etc/iofog-agent"
#          containerDestination: "test-mapping"
#          accessMode: "rw"

File: deploy/crds/k8s_v1alpha2_kog_crd.yaml (hunk 2 above)
Comment (SergeRadinovich) [reply deploy/crds/k8s_v1alpha2_kog_crd.yaml]: The text doesn't end up in the crd:
```
Name:         kogs.k8s.iofog.org
Namespace:    
Labels:       <none>
Annotations:  <none>
API Version:  apiextensions.k8s.io/v1beta1
Kind:         CustomResourceDefinition
Metadata:
  Creation Timestamp:  2019-09-04T21:56:23Z
  Generation:          1
  Resource Version:    18438786
  Self Link:           /apis/apiextensions.k8s.io/v1beta1/customresourcedefinitions/kogs.k8s.iofog.org
  UID:                 d5c3755f-cf5e-11e9-8538-42010aa80002
Spec:
  Additional Printer Columns:
    JSON Path:    .metadata.creationTimestamp
    Description:  CreationTimestamp is a timestamp representing the server time when this object was created. It is not guaranteed to be set in happens-before order across separate operations. Clients may not set this value. It is represented in RFC3339 form and is in UTC.

Populated by the system. Read-only. Null for lists. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata
    Name:  Age
    Type:  date
  Group:   k8s.iofog.org
  Names:
    Kind:       Kog
    List Kind:  KogList
    Plural:     kogs
    Singular:   kog
  Scope:        Namespaced
  Subresources:
    Status:
  Version:  v1alpha2
  Versions:
    Name:     v1alpha2
    Served:   true
    Storage:  true
Status:
  Accepted Names:
    Kind:       Kog
    List Kind:  KogList
    Plural:     kogs
    Singular:   kog
  Conditions:
    Last Transition Time:  2019-09-04T21:56:23Z
    Message:               no conflicts found
    Reason:                NoConflicts
    Status:                True
    Type:                  NamesAccepted
    Last Transition Time:  <nil>
    Message:               the initial names have been accepted
    Reason:                InitialNamesAccepted
    Status:                True
    Type:                  Established
  Stored Versions:
    v1alpha2
Events:  <none>
```

**Pull request merged**